### PR TITLE
Fix #73486 #73496 rewrite jumpToStartRepeat reorder conditions eval

### PR DESCRIPTION
**Pull request by ericfont**
I rewrote the loop in jumpToStartRepeat to handle conditions in proper order to fix a couple corner cases:
 * if ending a section with a single-measure repeat
 * if starting score with a single-measure section

Also I think it is cleaner and easier to read now.

Also added -j2 to ctest so travis runs tests with two CPUs.

### CODE REVIEW

File: libmscore/repeatlist.cpp [hunk 1]
```diff
@@ -399,24 +399,29 @@ void RepeatList::unwind()
...
-      // handle special case of end repeat (Measure m) has a section break
-      //
-      for (nm = m; nm && nm != _score->firstMeasure(); nm = nm->prevMeasure()) {
-            if (nm->repeatFlags() & Repeat::START || (nm->sectionBreak() && m != nm)) {
-                  if (nm->sectionBreak() && nm->nextMeasure())
-                        nm = nm->nextMeasure();
-                  break;
-                  }
-            }
+      // finalize the previous repeat segment
```
Comment (ericfont) [inline libmscore/repeatlist.cpp]: I moved the finalize of previous repeat segment to beginning of function, because it makes more sense to deal with it at the start, and also so I don't have to keep track of two measure variables in this function.

File: libmscore/repeatlist.cpp [hunk 2]
```diff
@@ -399,24 +399,29 @@ void RepeatList::unwind()
 
 Measure* RepeatList::jumpToStartRepeat(Measure* m)
       {
-      Measure* nm;
-      //
-      // go back to start of repeat or section break
-      // handle special case of end repeat (Measure m) has a section break
-      //
-      for (nm = m; nm && nm != _score->firstMeasure(); nm = nm->prevMeasure()) {
```
Comment (ericfont) [inline libmscore/repeatlist.cpp]: if you notice, I have the removed the test for nm being non-null.  The reason is that the only reason why nm would be null is if there is something truely corrupt with the score such that the links between measures is somehow broken going backwards before reaching firstMeasure.  So that is probably better left as a test for corruption somewhere else in mscore...and if there was such a corruption in the score, it would instead be better to stop and report an error to user.  Maybe in some seperate test score for broken links function.

**Comment by ericfont** [comment]
oops...let me fix this.

File: libmscore/repeatlist.cpp [hunk 3]
```diff
@@ -399,24 +399,29 @@ void RepeatList::unwind()
...
-                        nm = nm->nextMeasure();
-                  break;
-                  }
-            }
+      // finalize the previous repeat segment
       rs->len = m->tick() + m->ticks() - rs->tick;
       append(rs);
 
+      // search backwards until find start of repeat
+      while (true) {
```
Comment (ericfont) [inline libmscore/repeatlist.cpp]: I could have make this in to a big giant loop condition test with each condition placed in proper order, but I chose to instead break each condition apart because makes much more comprehendable.

File: libmscore/repeatlist.cpp [hunk 4]
```diff
@@ -399,24 +399,29 @@ void RepeatList::unwind()
...
+
+            if (m->repeatFlags() & Repeat::START)
+                  break;
+
+            if (m == _score->firstMeasure())
+                  break;
+
+            if (m->prevMeasure()->sectionBreak())
```
Comment (ericfont) [inline libmscore/repeatlist.cpp]: I much prefer the way I've wrote it here, where I first test if previous measure has section break before modifying the loop iterator variable.  The old loop method would have to undo the "nm = nm->prevMeasure()" with "nm = nm->nextMeasure()" if the section break occurs...and this not only is a little messy, but was the source of the error I was experiencing.

File: libmscore/repeatlist.cpp [hunk 5]
```diff
@@ -399,24 +399,29 @@ void RepeatList::unwind()
...
       rs->len = m->tick() + m->ticks() - rs->tick;
       append(rs);
 
+      // search backwards until find start of repeat
+      while (true) {
+
+            if (m->repeatFlags() & Repeat::START)
+                  break;
+
+            if (m == _score->firstMeasure())
```
Comment (ericfont) [inline libmscore/repeatlist.cpp]: because I perform this test & break before the next test which dereferences m->prevMeasure()->sectionBreak(), I am ensured that m->prevMeasure() exists.  Again, the only time when such event m->prevMeasure() not being valid would be if there was some broken link in the score's measure list.
